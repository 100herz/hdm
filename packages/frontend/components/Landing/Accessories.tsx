import Image from 'next/image'
import styled from '@emotion/styled'

export default function Accessories({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  return (
    <section id="accessories" className={`bg-gray-100 overflow-x-hidden lg:mt-24 ${className}`}>
      <BackgroundImage>
        <div className="container py-20">
          <header className="lg:mt-16 lg:ml-32">
            <h2 className="text-4xl font-bold lg:text-5xl">
              DAZU<sup className="font-normal">&reg;</sup>
            </h2>
            <p className="mt-4 text-xl leading-snug lg:text-2xl">Das Zubehörprogramm</p>
          </header>

          <div className="lg:justify-between lg:ml-32 lg:flex lg:mt-8">
            <div className="lg:w-1/3 lg:-mt-4 lg:mr-12">
              <div className="mt-12 lg:-mx-24">
                <Image
                  src="/images/hdm-dazu-zubehoer.jpg"
                  alt="Möbelbauplatten - modernen Regalböden und Möbelbauplatten"
                  width={768}
                  height={432}
                />
              </div>
            </div>

            <div className="lg:w-1/2">
              <p className="mt-12 lg:text-lg">
                Alles was zum Anbringen von Dekorpaneele, zur Bodenverlegung, zum Bau von Mobiliar oder dem dekorativen
                Innenausbau notwendig ist, hat Holz Design Moers im DAZU<sup>&reg;</sup>-Zubehörprogramm.
              </p>
              <p className="mt-12 lg:text-lg">
                Verlegung und Montage einfach gemacht - mit dem DAZU<sup>&reg;</sup>-Zubehörprogramm.
              </p>
            </div>
          </div>
        </div>
      </BackgroundImage>
    </section>
  )
}

const BackgroundImage = styled.div`
  @media (min-width: 1024px) {
    background-image: linear-gradient(rgba(247, 250, 252, 0.95), rgba(247, 250, 252, 0.95)), url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    background-position: left center;
    background-size: 100% auto;
  }
`
