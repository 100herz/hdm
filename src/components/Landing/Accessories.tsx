import styled from "styled-components"

const Grid = styled.div`
  display: grid;
  grid-template-areas:
    "grid-item-1"
    "grid-item-2";
  grid-template-columns: 1fr;

  @media (min-width: 768px) {
    grid-template-areas: "grid-item-1 grid-item-2";
    grid-template-columns: 1fr 1fr;
    column-gap: 8rem;
  }

  & > :nth-child(1) {
    grid-area: grid-item-1;
  }

  & > :nth-child(2) {
    grid-area: grid-item-2;
  }
`

const BackgroundImage = styled.div`
  @media (min-width: 768px) {
    background-image: linear-gradient(rgba(247, 250, 252, 0.95), rgba(247, 250, 252, 0.95)), url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    background-position: left bottom;
    background-size: 100% auto;
  }
`

type Props = {
  className?: string
}

const Accessories: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <section id="accessories" className={`lg:bg-gray-100 overflow-x-hidden lg:mt-24 ${className}`}>
      <BackgroundImage>
        <Grid className="container pt-20 lg:pb-20">
          <div className="lg:max-w-md">
            <header className="mt-16 lg:ml-32">
              <h2 className="text-4xl font-bold leading-tight lg:text-5xl">Regalböden / Möbelbauplatten</h2>
            </header>

            <picture className="block mt-12 lg:-mx-24">
              <source srcSet="/images/hdm-moebelbauplatten.webp" type="image/webp" />
              <source srcSet="/images/hdm-moebelbauplatten.jpg" type="image/jpeg" />
              <img
                src="/images/hdm-moebelbauplatten.webp"
                alt="Möbelbauplatten - modernen Regalböden und Möbelbauplatten"
              />
            </picture>

            <div className="">
              <p className="mt-12 lg:text-lg">
                Mit modernen Regalböden und Möbelbauplatten, in vielen Abmessungen und dem richtigen Zubehör, lässt sich
                einfach und schnell frischer Wind in alle Räume bringen.
              </p>
              <p className="mt-8 lg:text-lg">
                Ihrer Kreativität können Sie mit dem Holzzuschnitt-Programm von Holz Design Moers freien Lauf lassen.
              </p>
            </div>
          </div>

          <div className="lg:max-w-md">
            <header className="mt-16 lg:ml-32">
              <h2 className="text-4xl font-bold lg:text-5xl">
                DAZU<sup className="font-normal">&reg;</sup>
              </h2>
              <p className="mt-4 text-xl leading-snug lg:text-2xl">Das Zubehörprogramm</p>
            </header>

            <picture className="block mt-12 lg:-mx-24">
              <source srcSet="/images/hdm-dazu-zubehoer.webp" type="image/webp" />
              <source srcSet="/images/hdm-dazu-zubehoer.webp" type="image/jpeg" />
              <img src="/images/hdm-dazu-zubehoer.webp" alt="DAZU-Zubehörprogramm" />
            </picture>

            <div className="">
              <p className="mt-12 lg:text-lg">
                Alles was zum Anbringen von Dekorpaneele, zur Bodenverlegung, zum Bau von Mobiliar oder dem dekorativen
                Innenausbau notwendig ist, hat Holz Design Moers im DAZU<sup>&reg;</sup>-Zubehörprogramm. Verlegung und
                Montage einfach gemacht - mit dem DAZU<sup>&reg;</sup>-Zubehörprogramm.
              </p>
            </div>
          </div>
        </Grid>
      </BackgroundImage>
    </section>
  )
}

export default Accessories
