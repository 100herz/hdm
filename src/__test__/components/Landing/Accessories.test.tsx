import Accessories from "@/components/Landing/Accessories"
import { shallow, ShallowWrapper } from "enzyme"
import toJSON from "enzyme-to-json"

describe("Accessories Testings", () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<Accessories />)
  })

  test("matches snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
