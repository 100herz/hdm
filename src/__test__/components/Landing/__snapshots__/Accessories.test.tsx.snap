// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessories Testings matches snapshot 1`] = `
<section
  className="lg:bg-gray-100 overflow-x-hidden lg:mt-24 "
  id="accessories"
>
  <Accessories__BackgroundImage>
    <div
      className="container py-20"
    >
      <header
        className="mt-16 lg:ml-32"
      >
        <h2
          className="text-4xl font-bold lg:text-5xl"
        >
          DAZU
          <sup
            className="font-normal"
          >
            ®
          </sup>
        </h2>
        <p
          className="mt-4 text-xl leading-snug lg:text-2xl"
        >
          Das Zubehörprogramm
        </p>
      </header>
      <div
        className="lg:justify-between lg:ml-32 lg:flex lg:mt-8"
      >
        <div
          className="lg:w-1/3 lg:-mt-4 lg:mr-12"
        >
          <picture
            className="block mt-12 lg:-mx-24"
          >
            <source
              srcSet="/images/hdm-dazu-zubehoer.webp"
              type="image/webp"
            />
            <source
              srcSet="/images/hdm-dazu-zubehoer.jpg"
              type="image/jpeg"
            />
            <img
              alt="Möbelbauplatten - modernen Regalböden und Möbelbauplatten"
              src="/images/hdm-dazu-zubehoer.jpg"
            />
          </picture>
        </div>
        <div
          className="lg:w-1/2"
        >
          <p
            className="mt-12 lg:text-lg"
          >
            Alles was zum Anbringen von Dekorpaneele, zur Bodenverlegung, zum Bau von Mobiliar oder dem dekorativen Innenausbau notwendig ist, hat Holz Design Moers im DAZU
            <sup>
              ®
            </sup>
            -Zubehörprogramm.
          </p>
          <p
            className="mt-12 lg:text-lg"
          >
            Verlegung und Montage einfach gemacht - mit dem DAZU
            <sup>
              ®
            </sup>
            -Zubehörprogramm.
          </p>
        </div>
      </div>
    </div>
  </Accessories__BackgroundImage>
</section>
`;
